invenio_circulation/transitions/base.py

```diff
@@ -142,11 +143,34 @@ def ensure_item_is_available_for_checkout(self, loan):
             raise ItemNotAvailableError(item_pid=loan["item_pid"],
                                         transition=self.dest)
 
+    def ensure_item_is_available_for_checkout(self, loan):
```

Comment:
since this is used only in `ToItemOnLoan`, shall we move it inside there and not keep it in the base class?

Reply:
As this change will break a lot of test (due to the fact we move the function), I suggest to do it in a later step.